Add skeleton BBH pipeline

nilsvu opened the pull request:
## Proposed changes

Tools to `generate-id` > `start-inspiral` > `start-ringdown`. No automation yet and barely any smart parameter choices, but this should provide a good starting point to flesh out the transition from inspiral to ringdown. See https://github.com/sxs-collaboration/spectre/issues/5021 for details on future plans.

Try it like this:

```sh
./bin/spectre bbh generate-id -q 1 -D 20 -w 0.015 -o ./ID
# wait...
./bin/spectre bbh start-inspiral ./ID/InitialData.yaml -O ./Ev/
# wait...
./bin/spectre bbh start-ringdown ./Ev/Inspiral.yaml -O ./Ringdown/
```

(this won't really work yet because the inspiral won't finish properly, but that's #4600)

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu committed:
Add skeleton BBH pipeline

knelli2 commented on support/Pipelines/Bbh/InitialData.py:
```diff
@@ -0,0 +1,143 @@
+
+
+@click.command(name="generate-id", help=generate_id.__doc__)
+@click.option(
+    "--mass-ratio",
```
How do you feel about using another yaml to specify these? As we add more and more capability, having to write a new click option for every nob we want to tweak will probably get pretty annoying. This could simplify the command down to

```sh
spectre bbh generate-id InitialParams.yaml
```

where `InitialParams.yaml` is

```yaml
mass-ratio: 1.5
separation: 16
orbital-angular-velocity: 0.01
polynomial-order: 7
```

and then all the other parameters we didn't specify will be default.

This suggestion applies to all phases of the BBH pipeline.

nilsvu replied:
Ohh no, adding another "meta" input file seems like it could be a disaster :D I'd rather keep the knobs here down, and even remove the hp refinement options once we have AMR. If you want full control over options you can write your own input file and run it. The steps in the pipeline still work if you're starting them from a simulation that you didn't run with the pipeline.

nilsvu replied:
I would be fine with adding an option to supply the CLI option from a configuration file rather than on the command line if you want. We  already do something like that with the user config file and the status CLI. It's just an alternative way of passing the options to the CLI. There's a pretty generic way to do that with click.

knelli2 replied:
Yeah I think that would be nice.

knelli2 replied:
Did you want to make this change now or in a follow up PR?

nilsvu replied:
I wasn't planning to do this right away, since I don't have a use case right now

nilsdeppe replied:
I agree, could be nice, first need to see how all this works in practice :)

knelli2 commented on support/Pipelines/Bbh/InitialData.py:
```diff
@@ -0,0 +1,143 @@
+    "--refinement-level",
+    "-L",
+    type=int,
+    help="h-refinement level.",
+    default=0,
```
Not to be figured out for this PR, but do we want to allow defaults? We explicitly don't in our executable input files for a variety of reasons, and this sort of gets around that.

nilsvu replied:
My plan for these resolution options is to replace them with something like "lev" or "target truncation error" once we have AMR. I can remove the default here for now, no strong opinion. BTW a big reason for avoiding defaults in input files is that they're so hard to discover and can change your simulation in subtle ways. Here they're just a `--help` away. CLIs use defaults all the time (still have to be careful!).

knelli2 commented on support/Pipelines/Bbh/__init__.py:
```diff
@@ -0,0 +1,22 @@
+bbh_pipeline.add_command(start_inspiral_command)
+bbh_pipeline.add_command(start_ringdown_command)
+
+if __name__ == "__main__":
+    bbh_pipeline(help_option_names=["-h", "--help"])
```
Maybe print a pretty click warning somewhere that this is still very experimental and shouldn't be used for actual runs yet?

nilsvu replied:
It's not more experimental than the BBH input file that you would copy from tests/InputFiles :D

knelli2 replied:
It's not the input file that's experimental, it's the pipeline. My idea of a pipeline is smooth seamless transitions with little input required from the user. At the moment, this isn't that. But I hope it will be!

nilsvu replied:
Ah ok, agreed! Added some warnings.

knelli2 reviewed (commented):
I think this is a great start! We'll definitely need to tune input file parameters and such, but that can be done as we go.

nilsvu committed:
Fix cross-compilation in Dockerfile

For some reason this broke, maybe some package dependencies
changed upstream. Disabling recommended packages fixed it.

nilsvu committed:
Fix a test failure if filesystem has symlinks

knelli2 reviewed (commented):
This is a great start!

nilsvu commented:
Fixed the MPI failure (I hope)

nilsdeppe reviewed (dismissed):
I didn't read it all in fine detail, but this seems like a good way to structure and organize all this.

nilsvu committed:
Extend an instantiation to clang 15

Clang 15 on M2 macOS also fails to instantiate this class.

nilsvu commented:
Rebased
